File: exercicio10/palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool palindromo_recursivo(char *s, int inicio, int fim) {
    if (inicio >= fim)
        return true;
    
    if (s[inicio] != s[fim])
        return false;
    
    return palindromo_recursivo(s, inicio + 1, fim - 1);
}

bool palindromo(char *s) {
    int tamanho = strlen(s);
    return palindromo_recursivo(s, 0, tamanho - 1);
}

int main() {
    char s[] = "rada";

    if (palindromo(s)) {
        printf("'%s' é um palíndromo.\n", s);
    } else {
        printf("'%s' não é um palíndromo.\n", s);
    }

    return 0;
}

File: exercicio11/Palindromo.java
package exercicio11;

public class Palindromo {
    public static void main(String[] args) {
        String s = "teste";

        Palindromo_recursivo(s, 0, s.length() - 1);
    }

    public static boolean Palindromo_recursivo(String s, int inicio, int fim){
        if(inicio >= fim){
            return true;
        }
        if(s.charAt(inicio) != s.charAt(fim)){
            return false;
        }

        return Palindromo_recursivo(s, inicio + 1, fim - 1);

    }
}
